ui/components/MesheryPerformance/index.js

```diff
@@ -202,86 +225,85 @@ const styles = (theme) => ({
   },
 });
 
-class MesheryPerformanceComponent extends React.Component {
-  constructor(props) {
-    super(props);
-    const {
-      testName = '',
-      meshName = '',
-      url = '',
-      qps = '0',
-      c = '0',
-      t = '30s',
-      result,
-      staticPrometheusBoardConfig,
-      performanceProfileID,
-      profileName,
-      loadGenerator,
-      additional_options,
-      headers,
-      cookies,
-      reqBody,
-      contentType,
-      metadata,
-    } = props;
-
-    this.state = {
-      testName,
-      meshName,
-      url,
-      qps,
-      c,
-      t,
-      tValue: t,
-      loadGenerator: loadGenerator || 'fortio',
-      additional_options: additional_options || '',
-      result,
-      headers: headers || '',
-      cookies: cookies || '',
-      reqBody: reqBody || '',
-      contentType: contentType || '',
-      certificate: {},
-      certificateKey: {},
-      caCertificate: {},
-      profileName: profileName || '',
-      performanceProfileID: performanceProfileID || '',
-      timerDialogOpen: false,
-      blockRunTest: false,
-      urlError: false,
-      tError: '',
-      disableTest: !(URLValidator(url) || this.isJsonString(additional_options)),
-      testUUID: this.generateUUID(),
-      staticPrometheusBoardConfig,
-      selectedMesh: '',
-      availableAdapters: [],
-      availableSMPMeshes: [],
-      disableAvailableOptionsUploadButton: false,
-      disableAvailableOptionsInputField: false,
-      metadata,
-    };
-  }
-
-  isJsonString(str) {
+let eventStream = null;
+const MesheryPerformanceComponent = (props) => {
+  const {
+    testName = '',
+    meshName = '',
+    url = '',
+    qps = '0',
+    c = '0',
+    t = '30s',
+    result,
+    staticPrometheusBoardConfig,
+    performanceProfileID,
+    profileName,
+    loadGenerator,
+    additional_options,
+    headers,
+    cookies,
+    reqBody,
+    contentType,
+    metadata,
+  } = props;
+  const isJsonString = (str) => {
     try {
       JSON.parse(str);
     } catch (e) {
       return false;
     }
     return true;
-  }
-
-  handleChange = (name) => (event) => {
+  };
+  // Create individual state variables for each property
+  const [testNameState, setTestName] = useState(testName);
```

Comment:
You may consider combining states if there are states which contain related info